### ISSUE TITLE
Coverage is not working anymore because solidity-coverage bug changes state mutability

### ISSUE THREAD

**Issue by ottodevs** [aragon/aragon-apps#417]
To reproduce the error just clone the repo, run `npm install` and try `npm run coverage`

The errors I am getting seem like this:

```sh
aragon-apps/apps/voting/coverageEnv/contracts/Voting.sol:204:5:
TypeError: Overriding function changes state mutability from "pure" to "nonpayable".

function isForwarder() public returns (bool) 
 {__FunctionCoverageVoting('aragon-apps/apps/voting/contracts/Voting.sol',7);
```
The errors happen when solidity-coverage generates the instrumentation files.

[This was supposed to be fixed by solidity-coverage team](https://github.com/sc-forks/solidity-coverage/issues/131) but since aragon-apps is using a previous version I couldn't test it to make sure.

The current solution to make coverage working again seems to be removing decorators from functions, or updating solidity-coverage (this involves updating solc, therefore truffle, and seems more breaking solution than former one)

The last successful coverage report was generated on May: https://coveralls.io/github/aragon/aragon-apps

Here is an example of the errors I am talking about: https://travis-ci.org/aragon/aragon-apps/jobs/381949727 (search for "error" word in the page)

**Comment by ottodevs** [comment]
Coverage is actually working in other branches like `stacking-checkpointing` but these branches are using the same solidity-coverage versions, any ideas on what is happening with `master` branch then?

**Comment by sohkai** [comment]
Yes... this has been a constant pain point with solidity-coverage. I think bumping to 0.4.3 has worked for most of the other apps.

We'll also be bumping the truffle and solidity-coverage versions up soon, so this should be fixed with that.

**Comment by sohkai** [comment]
Will be fixed in #422.
